analytics/reports/netsuitesync.md

```diff
@@ -0,0 +1,593 @@
+       IFNULL(`QOH`, 0) AS `QOH variance`
+FROM
+  (SELECT oh.ORDER_ID,
+          oh.ORDER_NAME,
+          oh.ENTRY_DATE,
+          sale.PRODUCT_ID,
+          p.INTERNAL_NAME SKU,
+          sale.SALES,
+          variance.QOH,
+          variance.ATP,
+          variance.INVENTORY_ID
+   FROM order_header oh
+   JOIN
+     (SELECT oi.ORDER_ID,
+             SUM(oi.QUANTITY) SALES,
+             oi.PRODUCT_ID
+      FROM order_item oi
+      JOIN order_item_ship_group_assoc oisga ON oisga.ORDER_ID = oi.ORDER_ID
+      AND oisga.ORDER_ITEM_SEQ_ID = oi.ORDER_ITEM_SEQ_ID
+      JOIN order_item_ship_group oisg ON oisg.ORDER_ID = oisga.ORDER_ID
+      AND oisg.SHIP_GROUP_SEQ_ID = oisga.SHIP_GROUP_SEQ_ID
+      AND oisg.SHIPMENT_METHOD_TYPE_ID = 'POS_COMPLETED'
+      GROUP BY oi.ORDER_ID,
+               oi.PRODUCT_ID) sale ON sale.ORDER_ID = oh.ORDER_ID
+   AND oh.ENTRY_DATE > '2024-03-24'
+   AND oh.SALES_CHANNEL_ENUM_ID = 'POS_SALES_CHANNEL'
+   LEFT JOIN
+     (SELECT iiv.COMMENTS,
+             ii.PRODUCT_ID,
+             ii.INVENTORY_ITEM_ID INVENTORY_ID,
+             SUM(iiv.quantity_on_hand_var) QOH,
+             SUM(iiv.available_to_promise_var) ATP
+      FROM inventory_item_variance iiv
+      JOIN inventory_item ii ON ii.INVENTORY_ITEM_ID = iiv.INVENTORY_ITEM_ID
+      AND iiv.VARIANCE_REASON_ID = 'POS_SALE'
+      GROUP BY iiv.COMMENTS,
+               ii.PRODUCT_ID) variance ON variance.COMMENTS LIKE CONCAT('%', sale.ORDER_ID, '%')
+   AND variance.PRODUCT_ID = sale.PRODUCT_ID
+   JOIN product p on sale.PRODUCT_ID = p.PRODUCT_ID) AS virtual_table
+WHERE `SKU` != '9090909-999'
+  AND (((IF(`QOH`, `QOH`, 0) + `SALES`) <> 0))
+ORDER BY `ORDER_ID` DESC
+LIMIT 1000;
+```
+
+</details>
+
+### Query Logic
+
+**Data Selection:** The query selects data from multiple tables to gather information about POS orders, their sales quantities, and inventory variances. The key details retrieved include order ID, order name, entry date, SKU, sales quantity, and quantity on hand variance.
+
+**Filtering POS Orders:** The report focuses on POS orders by filtering for orders completed after a particular date, and belonging to the POS sales channel. This ensures the report includes only relevant transactions.
+
+**Calculating Sales and Variance:** The query calculates the total sales quantity for each order and product combination. It also sums the quantity on hand (QOH) and available to promise (ATP) variances from the inventory item variance table. The inventory variances are linked to the sales data based on comments containing the order ID.
+
+**Filtering Results:** The results are filtered to exclude a specific SKU and to highlight records where the sum of QOH variance and sales is not zero, indicating a discrepancy.
+
+**Ordering and Limiting Results:** The results are ordered by order ID in descending order to prioritize the most recent records. The output is limited to the first 1000 records.
+
+## POS Orders vs POS Variance Pie Chart
```

Comment:
not needed

Reply:
The report or Order and limiting results?

Reply:
the report, already added about it before, this one is only that in pie chart